starknet_py/net/gateway_client.py

```diff
@@ -67,13 +67,21 @@ def __init__(
         session: Optional[aiohttp.ClientSession] = None,
     ):
         """
+        .. deprecated:: 0.15.0
+            GatewayClient is deprecated and will not be supported in the future.
```

Comment:
Let's change it to something not so directly speaking about deprecation, something like _Gateway / Feeder Gateway API will become deprecated in the future. As a result, GatewayClient won't work and will eventually be removed. Consider migrating to FullNodeClient_